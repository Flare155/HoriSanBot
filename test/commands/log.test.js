import { expect, test, vi } from 'vitest';
import logCommand from '../../commands/global/log';

const createMockInteraction = (options = {}) => ({
    options: {
        getString: vi.fn((name) => options[name] || null),
    },
    reply: vi.fn(), // Mock reply method to capture responses
});

test('should fail on invalid time pattern', async () => {
    const interaction = createMockInteraction({
        medium: 'Watchtime',
        amount: '10',
        title: 'Test Title',
        notes: 'Some notes here',
        episode_length: '30',
    });
    await logCommand.execute(interaction);
    expect(interaction.reply).toMatchSnapshot();
});