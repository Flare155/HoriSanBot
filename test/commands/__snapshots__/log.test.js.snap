// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should fail on invalid time pattern 1`] = `
[MockFunction spy] {
  "calls": [
    [
      {
        "content": "\`‚ùå Invalid input format. Examples: \`2ep\`, \`1h3m\`. See \`/help log\` for more info.\`",
        "ephemeral": true,
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`should send embed after saving log 1`] = `
[MockFunction spy] {
  "calls": [
    [
      {
        "embeds": [
          {
            "color": 16252927,
            "description": "1 point/min ‚Üí +2 points",
            "fields": [
              {
                "inline": true,
                "name": "üìñ Title",
                "value": "Test Title",
              },
              {
                "inline": true,
                "name": "üìù Notes",
                "value": "Some notes here",
              },
            ],
            "footer": {
              "icon_url": "http://example.com/image.jpg",
              "text": "Keep up the great immersion!",
            },
            "thumbnail": {
              "url": "http://example.com/image.jpg",
            },
            "timestamp": "TIMESTAMP",
            "title": "üéâ test logged 2 minutes of Watchtime!",
          },
        ],
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
